rust-mode.el

```diff
@@ -711,6 +711,9 @@ Returns nil if the point is not within a Rust string."
      ;; Lifetimes like `'foo`
      (,(concat "'" (rust-re-grab rust-re-ident) "[^']") 1 font-lock-variable-name-face)
 
+     ;; variable name in for-loop
+     (,(concat "\\bfor[[:space:]]+" (rust-re-grab rust-re-ident) "[[:space:]]+in\\b") 1 font-lock-variable-name-face)
```

Comment:
The effect of that is that the highlighting of the identifier will not start before `in` has been inputted. Is this voluntary? If so, please add a test for that, otherwise, you can simplify the regex.

Reply:
@mookid yeah, I'm afraid of false-positives. Otherwise in something like this `for Ok(x) in expr()` the `Ok` would be highlighted as a variable.

Reply:
*(though this won't compile, but I think there's a way to pattern-match something in for-loop)*.

I'll add an additional testcase.

Reply:
> @mookid yeah, I'm afraid of false-positives. Otherwise in something like this `for Ok(x) in expr()` the `Ok` would be highlighted as a variable.

not sure about that, I got 
![image](https://user-images.githubusercontent.com/7757342/66130663-5c021f00-e5f2-11e9-85f1-1872185d2de9.png)

(which is not ideal, but not what you describe either) in any case.

Reply:
Hmm, okay… Yeah, thanks, I'll look then into simplifying the regex, and will add an additional testcase either way.

Reply:
@mookid thanks, comments addressed: I did both: simplified the regex and added another testcase.

As a side note: the variable font lock inside pattern-match, i.e. the `var` in `Foo{var}` probably have to be implemented somewhere separately as it isn't specific to a for-loop.

Reply:
Hmm… Please wait a bit, I thought about something… I think, in this regex shape I could actually just add `for` to the existing font-lock pattern, which is

```lisp
(,(concat "\\_<\\(?:let\\s-+ref\\|let\\|ref\\)\\s-+\\(?:mut\\s-+\\)?" (rust-re-grab rust-re-ident) "\\_>") 1 font-lock-variable-name-face)
```

let me try

Reply:
Okay, yeah, this seems to work as expected. Done.